modules/host/storageobligations.go

```diff
@@ -869,15 +869,38 @@ func (h *Host) StorageObligations() (sos []modules.StorageObligation) {
 			if err != nil {
 				return build.ExtendErr("unable to unmarshal storage obligation:", err)
 			}
+			statusString := map[storageObligationStatus]string{
+				obligationFailed:     modules.ObligationStatusFailed,
+				obligationRejected:   modules.ObligationStatusRejected,
+				obligationSucceeded:  modules.ObligationStatusSucceeded,
+				obligationUnresolved: modules.ObligationStatusUnresolved,
+			}[so.ObligationStatus]
+			if statusString == "" {
+				h.log.Severe("WARN: invalid obligation status encountered when reading database.")
+				return build.ExtendErr("unable to parse obligation status.", err)
```

Comment:
`err` will be `nil` here. My preference would just be to `return nil`. That way, the obligation is skipped, but others are still added. This keeps everything mostly-functional, but the log message will still indicate that something is wrong.